lib/common/socket.c

```diff
@@ -936,6 +936,8 @@ int32_t h2o_socket_getport(struct sockaddr *sa)
 static void create_ossl(h2o_socket_t *sock)
 {
     sock->ssl->ossl = SSL_new(sock->ssl->ssl_ctx);
+    /* set app data to be used in h2o_socket_ssl_new_session_cb */
+    assert(SSL_set_app_data(sock->ssl->ossl, sock));
```

Comment:
I think you need to split `assert` and the call to `SSL_set_app_data`. Otherwise, the call to the OpenSSL function will be eliminated when `NBEGUG` is set.